File: Src/java/elm/src/test/java/org/cqframework/cql/elm/IdObjectFactoryTest.java
package org.cqframework.cql.elm;

import java.lang.reflect.InvocationTargetException;
import java.util.Arrays;
import org.hl7.elm.r1.Element;
import org.junit.Test;

public class IdObjectFactoryTest {

    @Test
    public void ensureAllElementsHaveLocalId() {
        var factory = new IdObjectFactory();
        var methods = Arrays.asList(IdObjectFactory.class.getMethods()).stream()
                .filter(x -> Element.class.isAssignableFrom(x.getReturnType()));
        methods.forEach(x -> {
            try {
                Element e = (Element) x.invoke(factory);
                if (e.getLocalId() == null) {
                    throw new RuntimeException(
                            String.format("%s missing localId", e.getClass().getSimpleName()));
                }
            } catch (InvocationTargetException | IllegalAccessException e) {
                throw new RuntimeException(e);
            }
        });
    }
}

File: Src/java/elm/src/jvmTest/kotlin/org/cqframework/cql/elm/IdObjectFactoryTest.kt
package org.cqframework.cql.elm

import java.util.Locale
import kotlin.test.assertNotNull
import org.hl7.elm.r1.Element
import org.junit.jupiter.api.Test

internal class IdObjectFactoryTest {
    @Test
    fun ensureAllElementsHaveLocalId() {
        val factory = IdObjectFactory()
        val methods =
            IdObjectFactory::class.java.getMethods().filter {
                Element::class.java.isAssignableFrom(it.returnType)
            }
        methods.forEach { x ->
            val e = x.invoke(factory) as Element
            assertNotNull(
                e.localId,
                "%s returned null localId".format(Locale.US, e.javaClass.simpleName)
            )
        }
    }
}
